### PR TITLE
RedundantSyntax removes unnecessary string interpolator

### PR DESCRIPTION
**Pull request by gontard**
Idea from [gitter](https://gitter.im/scalacenter/scalafix?at=625756193ae95a1ec1e78751):

```diff
- println(s"Hi!")
+ println("Hi!")
```

### CODE REVIEW

File: scalafix-tests/output/src/main/scala/test/redundantSyntax/StringInterpolator.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  b = "foo"
+  b = "foo"
+  b = s"foo $a bar"
+
+  b = """foo"""
+  b =
+    """foo
+       |bar"""
+  b = s"""foo $a bar"""
+  b = s"""$a"""
```
Comment (gontard) [inline scalafix-tests/output/src/main/scala/test/redundantSyntax/StringInterpolator.scala:17]: If `a` is a `String`

```suggestion
  b = a
```

but i guess it's not doable with a syntactic rule

File: scalafix-tests/output/src/main/scala/test/redundantSyntax/StringInterpolator.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  b = "foo"
+  b = f"foo $a%2.2f"
+
+  b = raw"foo $a \nbar"
+  b = raw"foo\nbar"
+  b = raw"foo bar"
```
Comment (gontard) [inline scalafix-tests/output/src/main/scala/test/redundantSyntax/StringInterpolator.scala:24]: I found a solution to remove this one
```suggestion
  b = "foo bar"
```

with [a new condition in the rule](url):

```diff
-if config.stringInterpolator && (p == "s" || p == "f") =>
+if config.stringInterpolator && (p == "s" || p == "f" || (p == "raw" && StringContext.processEscapes(content) == content))
```

WDYT?

**Review by bjaglin — COMMENTED** [review]
:wave: thanks for putting this together!

Interestingly, like any generic syntactic rule, this could live in scalafmt, but it seems that it was never requested...

A good use-case for this rule would be to fail (and not rewrite) after writing by accident `println(s"variable: {variable})"`. Unfortunately, it's not possible to configure `scalafixOnCompile` to `--check` rather than rewrite (early feedback would be the best in that case). https://github.com/scalacenter/scalafix/pull/1375 could be extended to support this I guess, so if the need comes we can do it at the framework level rather than the rule one (adding an adhoc `lint` boolean would be cumbersome) anyway.

File: scalafix-rules/src/main/scala/scalafix/internal/rule/RedundantSyntaxConfig.scala [hunk 3]
```diff
@@ -6,7 +6,9 @@ import metaconfig.generic.Surface
 
 final case class RedundantSyntaxConfig(
     @Description("Remove final modifier from objects")
-    finalObject: Boolean = true
+    finalObject: Boolean = true,
+    @Description("Remove unnecessary string interpolator")
+    stringInterpolator: Boolean = true
```
[the remark this answers is not in the document]
Comment (github-sebastien-boulet) [reply scalafix-rules/src/main/scala/scalafix/internal/rule/RedundantSyntaxConfig.scala:11]: Yes i have thought about that too... and i came to the same conclusion.

File: docs/rules/RedundantSyntax.md [hunk 4]
```diff
@@ -4,13 +4,20 @@ id: RedundantSyntax
...
 - final object foo
 + object Foo
 ```
 
+```diff
+- println(s"Foo")
++ println("Foo")
+```
+
 Note: in Scala 2.12 and earlier removing the `final` modifier will slightly change the resulting bytecode - see [this bug ticket](https://github.com/scala/bug/issues/11094) for further information.
```
[the remark this answers is not in the document]
Comment (gontard) [reply docs/rules/RedundantSyntax.md]: I have reworked that.

File: scalafix-tests/input/src/main/scala/test/redundantSyntax/StringInterpolator.scala [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/*
+rules = RedundantSyntax
+RedundantSyntax.stringInterpolator = true
+*/
+
+package test.redundantSyntax
+
+class StringInterpolator {
```
[the remark this answers is not in the document]
Comment (gontard) [reply scalafix-tests/input/src/main/scala/test/redundantSyntax/StringInterpolator.scala:8]: ✅

File: scalafix-rules/src/main/scala/scalafix/internal/rule/RedundantSyntaxConfig.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (gontard) [reply scalafix-rules/src/main/scala/scalafix/internal/rule/RedundantSyntaxConfig.scala:11]: I thought about that too... but i came to the same conclusion

**Review by bjaglin — APPROVED** [review]
LGTM :+1:

Not merging right away to avoid updating the docs too much ahead of next release

**Pull request merged**